experimental/oak_baremetal_launcher/src/main.rs

```diff
@@ -18,51 +18,57 @@
 
 use anyhow::Context;
 use clap::Parser;
-use crosvm::Crosvm;
+use instance::LaunchedInstance;
+use instance_crosvm::CrosvmInstance;
+use instance_qemu::QemuInstance;
 use oak_baremetal_communication_channel::{
     client::{ClientChannelHandle, RequestEncoder},
     schema,
 };
-use qemu::Qemu;
 use std::{
     fs,
     io::{BufRead, BufReader},
     os::unix::net::UnixStream,
     path::PathBuf,
 };
 use tokio::signal;
-use vmm::{Params, Vmm};
 
-mod crosvm;
+mod instance;
+mod instance_crosvm;
+mod instance_qemu;
 mod lookup;
-mod qemu;
 mod server;
-mod vmm;
 
-#[derive(clap::ArgEnum, Clone, Debug, PartialEq)]
+/// Parameters used for launching VM instances
+#[derive(Parser, Clone, Debug, PartialEq)]
+pub struct VmParams {
+    /// Path to the VMM binary to execute.
+    #[clap(long, parse(from_os_str), validator = path_exists)]
+    pub vmm_binary: PathBuf,
+
+    /// Path to the binary to load into the VM.
+    #[clap(long, parse(from_os_str), validator = path_exists)]
+    pub app_binary: PathBuf,
+
+    /// Port to use for debugging with gdb
+    #[clap(long = "gdb")]
+    pub gdb: Option<u16>,
+}
+
+#[derive(clap::Subcommand, Clone, Debug, PartialEq)]
 enum Mode {
-    Qemu,
-    Crosvm,
+    /// Launch runtime in qemu
+    Qemu(VmParams),
+    /// Launch runtime in crosvm
+    Crosvm(VmParams),
 }
 
 #[derive(Parser, Debug)]
 struct Args {
-    /// Path to the `qemu-system-x86_64` binary.
-    #[clap(long, parse(from_os_str), validator = path_exists, default_value_os_t = PathBuf::from("/usr/bin/qemu-system-x86_64"))]
-    qemu: PathBuf,
-
-    /// Path to the `crosvm` binary.
-    #[clap(long, parse(from_os_str), validator = path_exists, default_value_os_t = PathBuf::from("/usr/local/cargo/bin/crosvm"))]
-    crosvm: PathBuf,
-
     /// Execution mode.
-    #[clap(arg_enum, long, default_value = "qemu")]
+    #[clap(subcommand)]
```

Comment:
Ooh, TIL about subcommand.

Reply:
It's pretty nice, imo we should use it more